Python/Product/ExternalProfilerDriver/VTuneInvoker.cs

```diff
@@ -0,0 +1,349 @@
+            {
+                candidatedirname = Path.Combine(Path.GetTempPath(), filename + "_" + date + "(" + count + ").vt");
+                count++;
+            }
+            Directory.CreateDirectory(candidatedirname);
+            _baseOutDir = candidatedirname;
+        }
+
+        public void AddCollectorSpec(VTuneCollectSpec collector)
+        {
+            CollectSpec = collector;
+        }
+
+        private static string NextResultDirInDir(string basedir)
+        {
+            if (!Directory.Exists(basedir))
+            {
+                throw new ArgumentException($"Expected directory {basedir} does not exist");
+            }
+
+            int latest = 0;
+            IEnumerable<string> previous = Directory.GetDirectories(basedir, "r*hs");
+            if (previous.Count() != 0)
+            {
+                latest = previous
+                           .Select(x => { var n = new FileInfo(x).Name; return n.Substring(1, n.Length - 3); })
+                           .Select(x => Int32.Parse(x))
+                           .Max();
+                latest += 1;
+            }
+
+            var latestReportName = "r" + latest.ToString("D3") + "hs"; // what happens if there's none?
+            return latestReportName;
+        }
+
+        public static string FreshDirectory(string baseName = "results")
+        {
+            string date = DateTime.Now.ToString("yyyyMMdd");
+            string outPath = Path.Combine(Path.GetTempPath(), baseName + "_" + date);
+
+            int count = 1;
+            while (File.Exists(outPath))
+            {
+                //outPath = Path.Combine(Path.GetTempPath(), baseName + "_" + date + "(" + count + ")");
+                outPath = Path.Combine(Path.GetTempPath(), $"{baseName}_{date} ({count})");
+                count++;
+            }
+            Directory.CreateDirectory(outPath);
+
+            return outPath;
+        }
+
+        public static string FreshResultDir(string basedir)
+        {
+            const string rdirtemplate = "r_{0:D4}";
+            if (basedir == "" || basedir == null || basedir == string.Empty)
+            {
+                basedir = Path.GetTempPath();
+            }
+            if (!Directory.Exists(basedir))
+            {
+                throw new ArgumentException($"Directory {basedir} does not exist");
+            }
+            int count = 1;
+            string candidate = string.Format(rdirtemplate, count);
+            while (Directory.Exists(Path.Combine(basedir, candidate)))
+            {
+                count++;
+                candidate = string.Format(rdirtemplate, count);
+            }
+            return candidate;
+        }
+    }
+
+    public class VTuneNotInstalledException : Exception
+    {
+        public override string Message
+        {
+            get
+            {
+                return "Only VTune 2017 or 2018 supported, see https://software.intel.com/en-us/intel-vtune-amplifier-xe";
```

Comment:
Always avoid including `en-us` in URLs. This page will automatically redirect to the user's language if it's just "....com/intel-vtune..."